code/WorkInProgress/MechanicMadness.dm

```diff
@@ -3,6 +3,242 @@
+			if(src.loc==user)
+				src.set_loc(get_turf(src))
+				user.drop_item()
+				return
+			return MouseDrop(user)
+		attack_self(mob/user as mob)
+			src.set_loc(get_turf(user))
+			user.drop_item()
+			return
+		MouseDrop(atom/target)
+		// thanks, whoever hardcoded that pick-up action into obj/item/MouseDrop()!
+			if(istype(target,/obj/screen/hud))
+				return
+			if(target.loc!=get_turf(target) && !isturf(target)) //return if dragged onto an item in another object (i.e backpacks on players)
+				return // you used to be able to pick up cabinets by dragging them to your backpack
+			return ..()
+	housing_handheld
+		var/obj/item/mechanics/trigger/trigger/the_trigger
+		slots=7 // One slot used by the permanent button
+		name="Device Frame"
+		desc="A massively shrunken component cabinet fitted with a handle and an external button. Due to the average mechanic's low arm strength, it only holds 6 components."
+		w_class = 3.0 // fits in backpacks but not pockets. no quickdraw honk boxess
+		density=0
+		anchored=0
+		icon_state="housing_handheld"
+		flags = FPRINT | EXTRADELAY | TABLEPASS | CONDUCT | ONBELT
+
+		spawn_contents=list(/obj/item/mechanics/trigger/trigger)
+		proc/get_trigger()
+			if (!istype(src.the_trigger))
+				src.the_trigger = (locate(/obj/item/mechanics/trigger/trigger) in src.contents)
+				if (!istype(src.the_trigger)) //no trigger?
+					for(var/obj/item in src.contents)
+						item.loc=get_turf(src) // kick out any mechcomp
+					qdel(src) // delet
+					return 0
+			return 1
+		attack_self(mob/user as mob)
+			if(src.open)
+				return ..() // you can just use the trigger manually from the UI
+			if(src.get_trigger() && !src.open && src.loc==user)
+				return src.the_trigger.attack_hand(user)
+			return
+
+/obj/item/mechanics/trigger/trigger // stolen code from the Button
+	name = "Device Trigger"
+	desc = "This component is the integral button of a device frame. It cannot be removed from the device. Can be used by clicking on the device when the device's cover is closed"
+	icon_state = "comp_button"
+	var/icon_up = "comp_button"
+	var/icon_down = "comp_button1"
+	density = 1
+	anchored= 1
+	level=1
+	w_class = 4
+	New()
+		..()
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ALLOW_MANUAL_SIGNAL)
+	attackby(obj/item/W as obj, mob/user as mob)
+		if(..()) return 1
+		return //attack_hand(user) // was causing issues
+
+	attack_hand(mob/user as mob)
+		..()
+		if (!istype(src.loc,/obj/item/storage/mechanics/housing_handheld))
+			qdel(src) //if outside the gun, delet
+			return
+		if(level == 1)
+			src.icon_state=icon_down
+			SPAWN_DBG(1 SECOND)
+				src.updateIcon()
+			SEND_SIGNAL(src,COMSIG_MECHCOMP_TRANSMIT_DEFAULT_MSG)
+			playsound(get_turf(src),'sound/machines/keypress.ogg',30)
+		else
+			qdel(src) // it's somehow been unanchored or something, kill it
+		return
+	afterattack(atom/target as mob|obj|turf|area, mob/user as mob)
+		qdel(src)// never should be outside of the gun (in someone's hands), so kill it
```

Comment:
no `..()`?

Reply:
It's not really ever supposed to be in someone's hands, so allowing it to hit things would be kindof silly.